packages/jest-resolve/src/__test__/resolve-test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+jsinfra
+ */
+
+'use strict';
+
+const ModuleMap = require('jest-haste-map').ModuleMap;
+const Resolver = require('../');
+
+describe('isCoreModule', () => {
+  it('returns false if `hasCoreModules` is false.', () => {
+    const moduleMap = new ModuleMap();
+    const resolver = new Resolver(moduleMap, {
+      browser: false,
+      hasCoreModules: false,
+    });
+    const isCore = resolver.isCoreModule('assert');
+    expect(isCore).toEqual(false);
+  });
+
+  it('returns true if `hasCoreModules` is false and `moduleName` is a core module.', () => {
```

Comment:
Wait, this doesn't make sense.

* If `hasCoreModules` is true, it should return true if you pass in a core module. If it is false, it should return false when passing in a core module.
* If `hasCoreModule` is true, it should return false if you pass in a non-core module. If it is false, it should always return false.